apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-dialog.tsx

```diff
@@ -0,0 +1,175 @@
+  DialogFooter,
+  Button
+} from '@harnessio/canary'
+import {
+  IInputDefinition,
+  RenderForm,
+  RootForm,
+  getTransformers,
+  outputTransformValues,
+  useZodValidationResolver
+} from '@harnessio/forms'
+import {
+  findPipeline,
+  getContent,
+  useCreateExecutionMutation,
+  useListBranchesQuery
+} from '@harnessio/code-service-client'
+import { PipelineParams } from '../pipeline-edit/context/PipelineStudioDataProvider'
+import { decodeGitContent, normalizeGitRef } from '../../utils/git-utils'
+import { createFormFromPipelineInputs } from './utils/utils'
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+
+const ADDITIONAL_INPUTS_PREFIX = '_'
+interface RunPipelineDialogProps {
+  open: boolean
+  onClose: () => void
+  pipelineId?: string
+  branch?: string
+}
+
+export default function RunPipelineDialog({ open, onClose, pipelineId, branch }: RunPipelineDialogProps) {
+  const { pipelineId: pipelineIdFromParams = '' } = useParams<PipelineParams>()
+  const navigate = useNavigate()
+
+  const [loading, setLoading] = useState(true)
+  const [pipeline, setPipeline] = useState<Record<string, unknown>>({})
+
+  const repoRef = useGetRepoRef()
+  const { data: listBranchesData, isLoading: listBranchesLoading } = useListBranchesQuery({
+    repo_ref: repoRef,
+    queryParams: {}
+  })
+
+  useEffect(() => {
+    findPipeline({ pipeline_identifier: pipelineId ?? pipelineIdFromParams, repo_ref: repoRef })
+      .then(pipelineData => {
+        getContent({
+          path: pipelineData.config_path ?? '',
+          repo_ref: repoRef,
+          queryParams: { git_ref: normalizeGitRef(branch ?? pipelineData.default_branch) ?? '', include_commit: true }
+        })
+          .then(pipelineFileContent => {
+            try {
+              const pipelineObj = parse(decodeGitContent(pipelineFileContent?.content?.data))
+              setPipeline(pipelineObj)
+            } catch (ex) {
+              // TODO: toast
+              console.error(ex)
+            }
+            setLoading(false)
+          })
+          .catch(e => {
+            throw e
+          })
+      })
+      .catch(ex => {
+        // TODO: toast
+        console.log(ex)
+        setLoading(false)
+      })
+  }, [pipelineId, repoRef])
+
+  const formDefinition = createFormFromPipelineInputs(pipeline)
+  const additionalInput: IInputDefinition[] = [
+    {
+      label: 'Branch',
+      inputType: InputType.select,
+      path: `${ADDITIONAL_INPUTS_PREFIX}.branch`,
+      inputConfig: {
+        options: listBranchesData?.map(branchItem => ({ label: branchItem?.name, value: branchItem?.name }))
+      }
+    },
+    { inputType: InputType.separator, path: '' }
+  ]
+  formDefinition.inputs = [...additionalInput, ...formDefinition.inputs]
+
+  const resolver = useZodValidationResolver(formDefinition)
+
+  const { mutateAsync: createExecutionAsync, isLoading: isLoadingCreateExecution } = useCreateExecutionMutation({})
+
+  const runPipeline = ({
```

Comment:
can we pull this out into a generic util function?
maybe the runtime input type can be a generic instead of `Record<string, unknown>`.

Reply:
added required pipeline types https://github.com/harness/canary/pull/334/files#diff-dcb989a7b34a33cbcb4c16fe09bef7380ee3c8132147dd22c30cb734d13d4e80